packages/runtime/container-runtime/src/deltaManagerRuntimeProxy.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	IDeltaManager,
+	IDeltaManagerEvents,
+	IDeltaQueue,
+	IDeltaSender,
+	ReadOnlyInfo,
+} from "@fluidframework/container-definitions";
+import { EventForwarder } from "@fluidframework/common-utils";
+import {
+	IClientConfiguration,
+	IClientDetails,
+	IDocumentMessage,
+	ISequencedDocumentMessage,
+	ISignalMessage,
+} from "@fluidframework/protocol-definitions";
+import { summarizerClientType } from "./summary";
+
+/**
+ * Proxy to the real IDeltaManager for the container runtime. It is used to restrict certain access to the runtime and
+ * layers below it.
+ */
+export class DeltaManagerRuntimeProxy
+	extends EventForwarder<IDeltaManagerEvents>
+	implements IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>
+{
+	public get IDeltaSender(): IDeltaSender {
```

Comment:
Can we create a base (abstract?) implementation of DM proxy, such that we do not need to reimplement these methods? And share this base class here and in the proxy that Loader uses?

Reply:
I was thinking the same thing. Let me do that.